Mismatched array bounds between function prototypes and definitions cause build to fail

```
common-lexers-helpers.c:567:32: error: argument 2 of type ‘const char *[555]’ with mismatched bound [-Werror=array-parameter=]
  567 | lexGet(State *  N, const char *tokenDescriptionArray[kCommonIrNodeTypeMax])
      |                    ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from common-lexers-helpers.c:58:
common-lexers-helpers.h:65:50: note: previously declared as ‘const char *[344]’
   65 | Token *         lexGet(State *  N, const char *  tokenDescriptionArray[kNoisyIrNodeTypeMax]);
      |                                    ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
common-lexers-helpers.c:627:51: error: argument 3 of type ‘const char *[]’ with mismatched bound [-Werror=array-parameter=]
  627 | lexPrintToken(State *  N, Token *  t, const char *tokenDescriptionArray[])
      |                                       ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
In file included from common-lexers-helpers.c:58:
common-lexers-helpers.h:68:69: note: previously declared as ‘const char *[344]’
   68 | void            lexPrintToken(State *  N, Token *  t, const char *  tokenDescriptionArray[kNoisyIrNodeTypeMax]);
      |                                                       ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
common-lexers-helpers.c:677:56: error: argument 3 of type ‘const char *[]’ with mismatched bound [-Werror=array-parameter=]
  677 | lexDebugPrintToken(State *  N, Token *  t, const char *tokenDescriptionArray[])
      |                                            ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
In file included from common-lexers-helpers.c:58:
common-lexers-helpers.h:69:74: note: previously declared as ‘const char *[344]’
   69 | void            lexDebugPrintToken(State *  N, Token *  t, const char *  tokenDescriptionArray[kNoisyIrNodeTypeMax]);
      |                                                            ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
common-lexers-helpers.c:740:75: error: argument 4 of type ‘const char *[]’ with mismatched bound [-Werror=array-parameter=]
  740 | lexPeekPrint(State *  N, int maxTokens, int formatCharacters, const char *tokenDescriptionArray[])
      |                                                               ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
In file included from common-lexers-helpers.c:58:
common-lexers-helpers.h:70:91: note: previously declared as ‘const char *[344]’
   70 | void            lexPeekPrint(State *  N, int maxTokens, int formatCharacters, const char *tokenDescriptionArray[kNoisyIrNodeTypeMax]);
      |                                                                               ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: note: unrecognized command-line option ‘-Wno-gnu-designator’ may have been intended to silence earlier diagnostics
cc1: all warnings being treated as errors

```

**Host OS (please complete the following information):**
 - OS: Linux 5.12.15-arch1-1 x86_64 GNU/Linux
 - gcc (GCC) 11.1.0

**You local changes (please complete the following information):**
- Output of `git diff`: 
```
```
- Output of `git remote -v`:
```
psm     git@github.com:phillipstanleymarbell/Noisy-lang-compiler.git (fetch)
psm     git@github.com:phillipstanleymarbell/Noisy-lang-compiler.git (push)
```

**Additional context**
Removing `-Wall -Werror` in `src/common/Makefile` bypasses the error. (Build still fails due to #537.)

--- a/src/common/common-lexers-helpers.c
+++ b/src/common/common-lexers-helpers.c
@@ -564,7 +564,7 @@ lexPut(State *  N, Token *  newToken)
 
 
 Token *
-lexGet(State *  N, const char *tokenDescriptionArray[kCommonIrNodeTypeMax])
+lexGet(State *  N, const char *  tokenDescriptionArray[kCommonIrNodeTypeMax])
 {
 	TimeStampTraceMacro(kCommonTimeStampKeyLexGet);
 
@@ -624,7 +624,7 @@ lexPeek(State *  N, int lookAhead)
 
 
 void
-lexPrintToken(State *  N, Token *  t, const char *tokenDescriptionArray[kCommonIrNodeTypeMax])
+lexPrintToken(State *  N, Token *  t, const char *  tokenDescriptionArray[kCommonIrNodeTypeMax])
 {
 	TimeStampTraceMacro(kCommonTimeStampKeyLexPrintToken);
 
@@ -674,7 +674,7 @@ lexPrintToken(State *  N, Token *  t, const char *tokenDescriptionArray[kCommonI
 }
 
 void
-lexDebugPrintToken(State *  N, Token *  t, const char *tokenDescriptionArray[kCommonIrNodeTypeMax])
+lexDebugPrintToken(State *  N, Token *  t, const char *  tokenDescriptionArray[kCommonIrNodeTypeMax])
 {
 	TimeStampTraceMacro(kCommonTimeStampKeyLexDebugPrintToken);
 
@@ -737,7 +737,7 @@ lexDebugPrintToken(State *  N, Token *  t, const char *tokenDescriptionArray[kCo
 }
 
 void
-lexPeekPrint(State *  N, int maxTokens, int formatCharacters, const char *tokenDescriptionArray[kCommonIrNodeTypeMax])
+lexPeekPrint(State *  N, int maxTokens, int formatCharacters, const char *  tokenDescriptionArray[kCommonIrNodeTypeMax])
 {
 	TimeStampTraceMacro(kCommonTimeStampKeyLexPeekPrint);
 

--- a/src/common/common-lexers-helpers.h
+++ b/src/common/common-lexers-helpers.h
@@ -67,4 +67,4 @@ Token *		lexPeek(State *  N, int lookAhead);
 
 void		lexPrintToken(State *  N, Token *  t, const char *  tokenDescriptionArray[kCommonIrNodeTypeMax]);
 void		lexDebugPrintToken(State *  N, Token *  t, const char *  tokenDescriptionArray[kCommonIrNodeTypeMax]);
-void		lexPeekPrint(State *  N, int maxTokens, int formatCharacters, const char *tokenDescriptionArray[kCommonIrNodeTypeMax]);
+void		lexPeekPrint(State *  N, int maxTokens, int formatCharacters, const char *  tokenDescriptionArray[kCommonIrNodeTypeMax]);

